usr/src/uts/common/fs/zfs/zap_micro.c

```diff
@@ -396,7 +396,7 @@ mzap_open(objset_t *os, uint64_t obj, dmu_buf_t *db)
 	 * it, because zap_lockdir() checks zap_ismicro without the lock
 	 * held.
 	 */
-	dmu_buf_init_user(&zap->zap_dbu, zap_evict, &zap->zap_dbuf);
+	dmu_buf_init_user(&zap->zap_dbu, NULL, zap_evict_async, &zap->zap_dbuf);
```

Comment:
To answer my own question: it isn't valid to have a sync evict func and not have an async evict func, because of how `dbuf_evict_user()` is implemented.  This is because the sync evict func probably free's the data structure containing the `dmu_buf_user_t`, and then `dbuf_evict_user()` dereferences it to determine if there is an async evict func.  I think this should be fixed.